pmd-core/src/main/java/net/sourceforge/pmd/rules/RuleFactory.java

```diff
@@ -121,16 +122,18 @@ public RuleReference decorateRule(Rule referencedRule, RuleSetReference ruleSetR
      * with regards to the expected schema, while RuleBuilder validates the semantics.
      *
      * @param ruleElement The rule element to parse
+     * @param resourceLoader The resource loader to load the rule from jar
      *
      * @return A new instance of the rule described by this element
      * @throws IllegalArgumentException if the element doesn't describe a valid rule.
      */
-    public Rule buildRule(Element ruleElement) {
+    public Rule buildRule(Element ruleElement, final ResourceLoader resourceLoader) {
```

Comment:
With the changes I explained above, this argument would then be removed and the RuleFactory has a instance field for the resource loader.